### ISSUE TITLE
When do we suppress rendering at the end of a gesture transition?

### ISSUE THREAD

**Issue by bokand** [bokand/gesture-vt#5]
In a regular cross document view transition, when a navigation starts, rendering is suppressed only once the navigation request receives a response and the UA is sure the new document will be swapped in. Until that happens, the page remains interactive and usable between the navigation starting and committing.

This doesn't seem like the right model for a gesture transition. In a gesture transition, the navigation begins at the end of the gesture but the end state of the transition is likely to be a screenshot representing the incoming page. It seems preferable to hold rendering at that state and make the page non-interactive - but there's a risk with slow navigation leaving users in a non-interactive state for too long.

**Comment by khushalsagar** [comment]
> It seems preferable to hold rendering at that state and make the page non-interactive

Yup. And the UA can deal with adding animations (like cross-fade) to deal with the screenshot mismatching the live web page once the navigation commits.

> but there's a risk with slow navigation leaving users in a non-interactive state for too long.

Hmmm, this is a hard one. IMO the best a UA can do is to add some overlay UI to indicate that the navigation is in progress and make it so the UI allows easily stopping the navigation.

**Comment by khushalsagar** [comment]
This made me think of https://github.com/bokand/gesture-vt/issues/8.

**Comment by bokand** [comment]
Re-thinking about this: I'm leaning away from suppressing rendering at all, or maybe suppress in the sense of paint holding during commit - same as MPA VT.

When the gesture finishes and we're waiting on network - we're still going to be displaying the view transition pseudos so the page will already be (mostly?) non-interactive. Since this is potentially a long period, it seems like good UX for the page to indicate work somehow, e.g subtle float animation or throbbing of the pseudo.

I suppose, as you mention, we could suppress rendering and the browser can be responsible for indicating work so maybe this is a question of how much flexibility do we give authors?

**Comment by khushalsagar** [comment]
> we're still going to be displaying the view transition pseudos so the page will already be (mostly?) non-interactive.

That's a very good point. We decided something similar for the duration of "update callback" for same-document transitions. The DOM doesn't align with the visuals so we don't hit-test and dispatch all events to the document root. Since that at least gives authors a way to detect that the user is trying to interact.

In this case, the visuals will align with the pseudo-tree we'll hit test on so there will be no accidental clicks to something you can't see.

Does this also mean we need to give the old Document an animation phase to animate away any UI that was waiting on the navigation to commit?
